### ISSUE TITLE
Allow for linking between config permutations

### ISSUE THREAD

**Issue by kpdecker** [walmartlabs/circus#7]
We should be able to pass in an array to the `Pack.config` API which will allow for multiple derivative builds to be generated based on different configuration parameters. To do this we will need to support linking such that:
1. Each permutation can have its own circus definition.
2. It's possible to link to either the same permutation (based on a config id value) or to a base package if no permutations are defined.

Basically the goal is to allow for builds that target a specific environment, say ios hybrid applications, and then can link to the same, if such a permutation exists.
